HeifImagePlugin.py

```diff
@@ -135,20 +138,30 @@ def _open(self):
             # We need to go deeper...
             if heif_file.color_profile['type'] in ('rICC', 'prof'):
                 self.info['icc_profile'] = heif_file.color_profile['data']
+        return heif_file
 
+    def _open(self):
         self.tile = []
-        self.heif_file = heif_file
+        self.heif_file = self._open_heif_file(Transformations is None)
 
     def load(self):
         heif_file, self.heif_file = self.heif_file, None
         if heif_file:
             try:
-                heif_file = heif_file.load()
+                try:
+                    heif_file = heif_file.load()
+                except HeifError as e:
+                    if not (e.code == 4 and e.subcode == 3003):
```

Comment:
Sorry for being annoying, but would it be possible to define constants/enum with meaningful naming?
```suggestion
                    if not (e.code == Errors.unsupported_color_conversion.code and e.subcode == Errors.unsupported_color_conversion.subcode):
```